packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -502,6 +517,28 @@ export class Function extends FunctionBase {
     const code = props.code.bind(this);
     verifyCodeConfig(code, props.runtime);
 
+    let profilingGroupEnvironmentVariables = {};
+    if (props.profilingGroup) {
```

Comment:
The logic here needs to change slightly.

Throw an error when profiling is set to false and profilingGroup is passed. 
OTOH, if profiling is false, ignore any configured profilingGroup. 
Also add corresponding unit tests.